kie-archetypes/kie-service-spring-boot-archetype/src/main/resources/META-INF/archetype-post-generate.groovy

```diff
@@ -0,0 +1,357 @@
+spring.datasource.url=jdbc:h2:./target/spring-boot-jbpm;MVCC=true
+spring.datasource.driver-class-name=org.h2.Driver
+spring.datasource.type=org.apache.tomcat.jdbc.pool.XADataSource
+
+#hibernate configuration
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.H2Dialect
+spring.jpa.properties.hibernate.show_sql=false
+spring.jpa.properties.hibernate.hbm2ddl.auto=update
+spring.jpa.hibernate.naming.physical-strategy=org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl
+
+#transaction manager configuration
+spring.jta.narayana.transaction-manager-id=1
+""";
+
+def BPMDatabaseConfigMySQL = """
+#data source configuration
+spring.datasource.username=jbpm
+spring.datasource.password=jbpm
+spring.datasource.url=jdbc:mysql://localhost:3306/jbpm
+spring.datasource.driver-class-name=com.mysql.jdbc.jdbc2.optional.MysqlXADataSource
+spring.datasource.type=org.apache.tomcat.jdbc.pool.XADataSource
+
+#hibernate configuration
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL5InnoDBDialect
+spring.jpa.properties.hibernate.show_sql=false
+spring.jpa.properties.hibernate.hbm2ddl.auto=update
+spring.jpa.hibernate.naming.physical-strategy=org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl
+
+#transaction manager configuration
+spring.jta.narayana.transaction-manager-id=1
+""";
+
+def BPMDatabaseConfigPostgres = """
+#data source configuration
+spring.datasource.username=jbpm
+spring.datasource.password=jbpm
+spring.datasource.url=jdbc:postgresql://localhost:5432/jbpm
+spring.datasource.driver-class-name=org.postgresql.xa.PGXADataSource
+spring.datasource.type=org.apache.tomcat.jdbc.pool.XADataSource
+
+#hibernate configuration
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
+spring.jpa.properties.hibernate.show_sql=false
+spring.jpa.properties.hibernate.hbm2ddl.auto=update
+spring.jpa.hibernate.naming.physical-strategy=org.hibernate.boot.model.naming.PhysicalNamingStrategyStandardImpl
+
+#transaction manager configuration
+spring.jta.narayana.transaction-manager-id=1
+""";
+
+println "Updating app configuration for app type: " + appType;
+if( appType == "bpm" ) {
+    println "Updating application properties...";
+    def appPropertiesContent = appPropertiesFile.getText('UTF-8');
+    def devAppPropertiesContent = devAppPropertiesFile.getText('UTF-8');
+
+    println "- adding server capabilities";
+    appPropertiesContent = appPropertiesContent.replace(kieServerCapabilitiesMarker, serverCapabilitiesBPM);
+    devAppPropertiesContent = devAppPropertiesContent.replace(kieServerCapabilitiesMarker, serverCapabilitiesBPM);
+
+    println "- adding jbpm configuration";
+    appPropertiesContent = appPropertiesContent.replace(jbpmConfigMarker, BPMJBPMConfig);
+    devAppPropertiesContent = devAppPropertiesContent.replace(jbpmConfigMarker, BPMJBPMConfig);
+
+    println "- adding database configuration";
+    println "   - using dbType: " + dbType;
+    if( dbType == "h2" ) {
```

Comment:
I would actually include all the them in pom.xml as profiles so users:
- have the profiles in every application
- they have a complete set of information (driver, application.properties setup) in the app in case they would like to use another database
- less complexity in the archetype itself
- less options to chose from when generating app